nucypher/network/nodes.py

```diff
@@ -1002,7 +1002,7 @@ def _worker_is_bonded_to_staker(self) -> bool:
         """
         staker_address = self.staking_agent.get_staker_from_worker(worker_address=self.worker_address)
         if staker_address == BlockchainInterface.NULL_ADDRESS:
-            raise self.DetachedWorker
+            raise self.DetachedWorker(f"Worker {self.worker_address} is detached")
```

Comment:
Hmmm, not sure I like this error string. It's certainly an improvement over the previous, but I don't feel anymore enlightened after reading it.

Reply:
It helps if you get the exception or read the associated log and try to figure out which worker address was involved.